### DISCUSSION TITLE
Is handeling evething in one route antipattern?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [nitrojs/nitro#2488] [General]
I have some specific use cases where I serve multiple things with nitro, one of them being a nextjs app where I manually handle next and give it to the nitro eventHandler, because it would simplify things a lot I wanted to ask if it is an antipattern to have everything in one spread `[...].ts`route handled and if it impacts performance? If there is something I could do better I would love to hear suggestions. Thanks.

````js
const dev = process.env.NODE_ENV !== "production";
const app = next({ dev, dir: "./ui/nextjs" });
const handle = app.getRequestHandler();

export default eventHandler((event) => {

  if(event.path.includes("admin")) {
    return app
    .prepare()
    .then(() => {
      return handle(event.node.req, event.node.res);
    })
    .catch((error) => {
      throw error;
    });  
  }
  
  if(event.path === "/") {
    // return something different
  }

  if(event.path === "/xyz") {
    // return something for xyz
  }
});

```

**Comment by airtonix** [comment]
yes.

question asked another way: 

- Are god components an anti pattern? 
- Why are more files rather than less files useful in software projects?
